.github/workflows/deploy-prod.yaml

```
@@ -0,0 +1,65 @@
+name: Deploy to production
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  build-test-and-deploy-production:
+    name: Build and deploy to production environment on merge to master branch
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Setup gcloud CLI
+        uses: google-github-actions/setup-gcloud@master
+        with:
+          service_account_email: ${{ secrets.GCP_SA_DIGDIR_FDK_GCR_EMAIL }}
+          service_account_key: ${{ secrets.GCP_SA_DIGDIR_FDK_GCR_KEY }}
+
+      - name: Configure docker to use the gcloud command-line tool as a credential helper
+        run: gcloud auth configure-docker
+
+      - name: Build Docker image
+        run: docker build . -t eu.gcr.io/digdir-fdk-infra/fdk-cms-service:$GITHUB_SHA -t eu.gcr.io/digdir-fdk-infra/fdk-cms-service:latest --build-arg GITHUB_TOKEN=${{ secrets.GITHUB_TOKEN }}
+
+      - name: Publish docker image to Google Cloud
+        run: docker -- push eu.gcr.io/digdir-fdk-infra/fdk-cms-service -a
```

```suggestion
      - name: Set up Docker Buildx
        id: buildx
        uses: docker/setup-buildx-action@v1

      - name: Build and push
        id: docker_build
        uses: docker/build-push-action@v2
        with:
          context: ./
          file: ./Dockerfile
          push: true
          tags: |
            eu.gcr.io/digdir-fdk-infra/fdk-cms-service:$GITHUB_SHA
            eu.gcr.io/digdir-fdk-infra/fdk-cms-service:latest
          cache-from: type=local,src=/tmp/.buildx-cache
          cache-to: type=local,dest=/tmp/.buildx-cache
          build-args:  |
            GITHUB_TOKEN=${{ secrets.GITHUB_TOKEN }}
```

Hva med å bruke docker sine Github Actions?

Kan potensielt legge inn litt cache også :)

```
      - name: Cache Docker layers
        uses: actions/cache@v2.1.5
        with:
          path: /tmp/.buildx-cache
          key: ${{ runner.os }}-buildx-${{ github.sha }}
          restore-keys: |
            ${{ runner.os }}-buildx-
```

---

Liker forslagene til endring, ta de gjerne med. Cache har stått på plakaten lenge

---

```suggestion
      - name: Cache Docker layers
        uses: actions/cache@v2.1.5
        with:
          path: /tmp/.buildx-cache
          key: ${{ runner.os }}-buildx-${{ github.sha }}
          restore-keys: |
            ${{ runner.os }}-buildx-
      - name: Set up Docker Buildx
        id: buildx
        uses: docker/setup-buildx-action@v1

      - name: Build and push
        id: docker_build
        uses: docker/build-push-action@v2
        with:
          context: ./
          file: ./Dockerfile
          push: true
          tags: |
            eu.gcr.io/digdir-fdk-infra/fdk-cms-service:$GITHUB_SHA
            eu.gcr.io/digdir-fdk-infra/fdk-cms-service:latest
          cache-from: type=local,src=/tmp/.buildx-cache
          cache-to: type=local,dest=/tmp/.buildx-cache
          build-args:  |
            GITHUB_TOKEN=${{ secrets.GITHUB_TOKEN }}
```